tests/test_init.py

```diff
@@ -3,26 +3,196 @@
+    ):
+        """
+        Check that :command:`systemd-detect-virt` correctly detects the container runtime
+        """
+        output = auto_container.connection.run_expect(
+            [0], "systemd-detect-virt"
+        ).stdout
+        runtime = container_runtime.runner_binary
+        assert (
+            runtime in output
+        ), f"systemd-detect-virt failed to detect {runtime}"
+
+    def test_journald(self, auto_container):
+        """
+        Check that :command:`journald` behaves correctly
+        """
+
+        # Check that we reached at least the multiuser target
+        journal = auto_container.connection.run_expect(
+            [0], "journalctl --boot"
+        )
+        assert (
+            "Reached target Multi-User System" in journal.stdout
+        ), "Multi-User target was not reached"
+
+    def test_hostnamectl(self, auto_container, host, container_runtime):
+        """
+        Check expected values from :command:`hostnamectl`
+        """
+
+        hostnamectl = auto_container.connection.run_expect([0], "hostnamectl")
+        # Process the printed values to a string map
+        values = TestSystemd._split_values(hostnamectl.stdout)
+        assert values["Chassis"] == "container", "Chassis mismatch"
+        assert (
+            "SUSE Linux Enterprise Server" in values["Operating System"]
+        ), "Missing SUSE tag in Operating system"
+        assert (
+            values["Virtualization"] == container_runtime.runner_binary
+        ), "Virtualization tag mismatch"
+
+    def test_timedatectl(self, auto_container):
+        """
+        Check that :command:`timedatectl` can be invoked without issues
+        """
+        output = auto_container.connection.run_expect(
+            [0], "timedatectl"
+        ).stdout
+        assert "Time zone: UTC" in output, "Time zone not set to UTC"
+
+        # Check that the reported timestamp for UTC and local time match the system time
+        def check_timestamp(pattern, timestamp, delta):
+            """Checks the timedatectl output for the given pattern against the given timestamp
+            e.g. use the "Universal time" as pattern and datetime.utcnow() to check for the UTC time
+            """
+            grep = [line for line in output.split("\n") if pattern in line]
+            assert (
+                len(grep) == 1
+            ), f"{pattern} not present in timedatectl output"
+            tsp = (
+                grep[0].strip()[len(pattern) + 2 :].strip()
+            )  # Extract actual timestamp
+            tsp = datetime.datetime.strptime(tsp, "%a %Y-%m-%d %H:%M:%S UTC")
+            print(
+                "%s - %s"
+                % (
+                    tsp.strftime("%m/%d/%Y, %H:%M:%S"),
+                    timestamp.strftime("%m/%d/%Y, %H:%M:%S"),
+                )
+            )
+            # Ensure the time difference is the higher - lower
+            diff = (tsp - timestamp) if tsp > timestamp else (timestamp - tsp)
```

Comment:
Did you just reinvent `math.abs()` :wink:?

Reply:
I wish it would have been that simple. Without this I sometimes got weird constructs like day `-1` which makes the following comparison fail.

Reply:
couldn't it be as by Dan suggestion ? (import math):
` diff = math.abs(tsp - timestamp)`

Reply:
No because it's an instance of `datetime.Timedelta` and not a float.

Reply:
Example:

```python
import datetime
import time

t1 = datetime.datetime.now()
time.sleep(2)
t2 = datetime.datetime.now()
print(t1-t2) # -1 day, 23:59:58.999138
print(t2-t1) # 0:00:01.000862

d1 = t1-t2
d2 = t2-t1

delta = datetime.timedelta(seconds=1)
print(d1 < delta) # True
print(d2 < delta) # False
```

Reply:
yes, anyway abs is Supported Operation od datetime:https://docs.python.org/3/library/datetime.html#datetime.timedelta

Reply:
@grisu48 jfyi as @m-dati correctly noted: `abs(d1) == abs(d2)` is True

Reply:
Oh nice, this works.

I tried it first with `math.abs` and that does not work, however `abs` does indeed what it should be doing.